src/main/java/edu/umich/ctools/sectionsUtilityTool/FriendServlet.java

```diff
@@ -0,0 +1,212 @@
+		request.setCharacterEncoding("UTF-8");
+		M_log.debug("friendRestApiCall(): called");
+		Friend myFriend = new Friend();
+		PrintWriter out = response.getWriter();
+		response.setContentType("application/json");
+		Properties appExtSecureProperties = Friend.appExtSecurePropertiesFile; 
+		if(appExtSecureProperties==null) {
+			response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+			out = response.getWriter();
+			out.print(props.getString("property.file.load.error"));
+			out.flush();
+			M_log.error("Failed to load system properties(sectionsToolProps.properties) for SectionsTool");
+			return;
+		}
+		friendApiConnectionLogic(request,response, myFriend);
+	}	
+
+	/*
+	 * This function has logic that execute client(i.e., browser) request and get results from the Friend Accounts  
+	 */
+	private void friendApiConnectionLogic(HttpServletRequest request, HttpServletResponse response, Friend myFriend)
+			throws IOException {
+		String pathInfo = request.getPathInfo();
+		PrintWriter out = response.getWriter();
+		JsonObject json;
+		switch (pathInfo){
+		case "/friendCreate":
+			String friendInviteEmail = request.getParameter(PARAMETER_ID);
+			String friendInstructorEmail = request.getParameter(PARAMETER_INSTRUCTOR_EMAIL);
+			String friendInstructorFirstName = request.getParameter(PARAMETER_INSTRUCTOR_FIRST_NAME);
+			String friendInstructorLastName = request.getParameter(PARAMETER_INSTRUCTOR_LAST_NAME);
+			if(friendInviteEmail == null || 
+					friendInstructorEmail == null ||
+					friendInstructorFirstName == null || 
+					friendInstructorLastName == null){
+				response.setStatus(400);
+				json = Json.createObjectBuilder()
+						.add("message", "request error: required parameters for request not specified")
+						.build();
+			}
+			else{
+				json = doFriendLogic(myFriend, 
+						friendInviteEmail, 
+						friendInstructorEmail, 
+						friendInstructorFirstName, 
+						friendInstructorLastName);
+			}
+			break;
+		default:
+			response.setStatus(400);
+			json = Json.createObjectBuilder()
+					.add("message", "request error")
+					.build();
+			break;
+		}
+		out.print(json.toString());
+		out.flush();
+	}
+	
+	private JsonObject buildResponseObject(String message, String detailedMessage){
+		return Json.createObjectBuilder()
+				.add("message", message)
+				.add("detailedMessage", detailedMessage)
+				.add("FriendURL", Friend.friendUrl)
+				.build();
+	}
+
+	private JsonObject doFriendLogic(Friend myFriend, 
+			String inviteEmail, 
+			String instructorEmail, 
+			String instructorFirstName, 
+			String instructorLastName){
+		JsonObject json;
+		String instructorName = instructorFirstName + " " + instructorLastName;
+		
+		HashMap<String, String> friendValues = new HashMap<String,String>();
+		
+		friendValues.put("inviteEmail", inviteEmail);
+		friendValues.put("instructorEmail", instructorEmail);
+		friendValues.put("instructorName", instructorName);
+		
+		//Step 1: determine if friend exists
```

Comment:
The later steps come in the next calls. I still thought it was a good idea to comment the steps.
1. determine if the friend exists
2. create the account
3. notify the instructor